internal/storage/local_chunk_manager.go

```diff
@@ -18,82 +18,145 @@ package storage
 
 import (
 	"errors"
+	"io"
 	"io/ioutil"
 	"os"
 	"path"
+	"path/filepath"
+	"strings"
 
 	"golang.org/x/exp/mmap"
 
 	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/util/errorutil"
 )
 
 // LocalChunkManager is responsible for read and write local file.
 type LocalChunkManager struct {
 	localPath string
 }
 
+var _ ChunkManager = (*LocalChunkManager)(nil)
+
 // NewLocalChunkManager create a new local manager object.
-func NewLocalChunkManager(localPath string) *LocalChunkManager {
+func NewLocalChunkManager(opts ...Option) *LocalChunkManager {
```

Comment:
when to clean the data on local chunk manager?
what if local chunk manager disk full?

Reply:
I would suggest to implement an LRU policy and set a size to evict local chunk, but that could be on other patch.
But the local cache should be cleaned I guess, especially on compaction and close.